package com.example.dictionary.feature_dictionary.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.dictionary.feature_dictionary.domain.model.Meaning
import com.example.dictionary.feature_dictionary.domain.model.WordInfo

@Entity
data class WordInfoEntity(
    @PrimaryKey val word: String,
    val phonetic: String?,
    val origin: String?,
    val meanings: List<Meaning>,
    val audioUrl: String? = null
) {
    fun toWordInfo(): WordInfo {
        return WordInfo(
            meanings = meanings,
            word = word,
            origin = origin ?: "",
            phonetic = phonetic ?: "",
            audioUrl = audioUrl
        )
    }
}

